### ISSUE TITLE
Logging Gzipped responses in HttpLoggingInterceptor as a NetworkInterceptor

### ISSUE THREAD

**Issue by mattsheppard** [square/okhttp#3780]
I'd like to be able to have HTTP Logging which includes redirect requests (i.e. it must be a networkInterceptor) and also logs the uncompressed version of any gzipped response bodies, which HttpLoggingInterceptor does not do if it's being used as a networkInterceptor.

If I were to implement support for gunzipping and logging response bodies when HttpLoggingInterceptor is used as a networkInterceptor is that a change you'd likely merge, or am I better off just forking my own HttpLoggingInterceptor?

Note: I saw https://github.com/square/okhttp/issues/1955 was a similar request, but it seems to have been closed because the original reporter was happy to run it a normal interceptor to get the gzipped body logged.

**Comment by JakeWharton** [comment]
I think I'm okay with this (as a small part of the OkHttp team). The
biggest problem is that it adds more complexity, but I have convinced
myself it's worthwhile as I usually install the logging interceptor as a
network interceptor and miss out on transparent gzip-compressed responses.
I'd like to see both the compressed and uncompressed size in the log and
maybe even a percentage difference between the two.

I'd wait for Jesse to chime in though before starting any PR.

**Comment by swankjesse** [comment]
Are the bodies of redirects ever interesting? In either case, sure, please send a PR.

**Comment by mattsheppard** [comment]
> Are the bodies of redirects ever interesting

No, probably not - Maybe it's clearer if I say...

1. If I install HttpLoggingInterceptor as a networkInterceptor currently, I see the redirects but not gzipped body of the last (or any) response.

2. If I install HttpLoggingInterceptor as an interceptor currently, I see the decompressed version of the gzipped body of any responses which have them but no redirects.

I want the best of both worlds - Redirects and gzipped bodies both logged. I could install HttpLoggingInterceptor twice, but the resulting interleaving (or having two seperate logs) makes it hard to follow.

Anyhow, I've got a hacked up copy of HttpLoggingInterceptor which does the the gunzipping already, so I'll try to clean it up and submit it soon!

**Comment by swankjesse** [comment]
Done!
